### PR TITLE
Extract SchoolAutocompleteDropdown (from census-styled label) and add tests

### PR DESCRIPTION
**Pull request by aoby**

### CODE REVIEW

File: apps/Gruntfile.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ module.exports = function (grunt) {
       `require('${path.resolve(process.env.mocha_entry)}');`;
     const file = `// Auto-generated
 import 'babel-polyfill';
+import 'whatwg-fetch';
 import { throwOnConsoleErrorsEverywhere, throwOnConsoleWarningsEverywhere } from './util/testUtils';
```
Comment (Erin007) [inline apps/Gruntfile.js:28]: `whatwg-fetch` is also referenced here: https://github.com/code-dot-org/code-dot-org/blob/621db84cc5e1a7b2506c449d4a89d15d68697c75/apps/Gruntfile.js#L542. We need it in both places?

**Comment by Erin007** [comment]
cc: @drewsamnick This change shouldn't affect how we're currently using the SchoolAutocomplete dropdown on /yourschool or HoC signup, just want to keep you in the loop.

File: apps/Gruntfile.js (hunk 1 above)
Comment (islemaster) [reply apps/Gruntfile.js:28]: Yes - line 542 in the gruntfile makes sure fetch is polyfilled in our built bundles.  Line 27 adds it to the entry tests template, so it's also available when running a subset of unit tests.

File: apps/src/templates/SchoolAutocompleteDropdown.jsx [hunk 2]
```diff
@@ -18,10 +16,13 @@ export default class SchoolAutocompleteDropdown extends Component {
     if (q.length < 4) {
       return Promise.resolve();
     }
-    return fetch(`/dashboardapi/v1/schoolsearch/${encodeURIComponent(q)}/40`)
-      .then(response => response.ok ? response.json() : [])
+    const url = `/dashboardapi/v1/schoolsearch/${encodeURIComponent(q)}/40`;
+    return fetch(url)
+      .then(response => {
+        return response.ok ? response.json() : [];
+      })
```
Comment (islemaster) [inline apps/src/templates/SchoolAutocompleteDropdown.jsx]: This is a good spot for an arrow function expression.
```js
.then(response => response.ok ? response.json() : [])
```

File: apps/test/unit/templates/SchoolAutocompleteDropdownTest.js [hunk 3]
```diff
@@ -1,15 +1,102 @@
...
+        onChange={handleChange}
       />
     );
-    expect(wrapper.find('VirtualizedSelect').exists()).to.be.true;
+
+    select = schoolAutocompleteDropdown.find('VirtualizedSelect');
+  });
+
+  it('renders VirtualizedSelect', () => {
+    expect(select.exists()).to.be.true;
```
Comment (islemaster) [inline apps/test/unit/templates/SchoolAutocompleteDropdownTest.js]: We use [`chai-enzyme`](https://github.com/producthunt/chai-enzyme) which lets you write:
```js
expect(select).to.exist;
```
[(docs)](https://github.com/producthunt/chai-enzyme#present)

File: apps/test/unit/templates/SchoolAutocompleteDropdownTest.js [hunk 4]
```diff
@@ -1,15 +1,102 @@
...
+  });
+
+  it('Displays supplied value', () => {
+    expect(select.prop('value')).to.equal("12345");
```
Comment (islemaster) [inline apps/test/unit/templates/SchoolAutocompleteDropdownTest.js]:
```js
expect(select).to.have.prop('value', '12345');
```
[(docs)](https://github.com/producthunt/chai-enzyme#propkey-val)

**Pull request merged**